### PR TITLE
Build on Semaphore

### PR DESCRIPTION
**Pull request by lemald**
#### Summary of changes
**Asana Ticket:** [💳 Build socket-proxy on Semaphore](https://app.asana.com/0/584764604969369/1197425148490073/f)

Ticket said to use asdf, and the best example of that I could find was actually in the API repo's `semaphore/setup.sh`. As for the overall structure of the rest of the individual steps, I mostly followed the example of Arrow and had separate parallel scripts rather than one `static_analysis.sh`. Most of the scripts are really simple, but `dialyzer.sh` is copied from Arrow and does some fancy stuff to play along nicely with the Semaphore cache.

The reviewer should also look at the project settings for `socket_proxy` on Semaphore.

#### Reviewer Checklist
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] Meets ticket's acceptance criteria

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (fjlanasa) [inline config/dev.exs:1]: can revert

Comment (lemald) [reply config/dev.exs:1]: Unfortunately `mix format --check-formatted` didn't like the empty file without the newline, hence the change.

Comment (fjlanasa) [reply config/dev.exs:1]: ah of course

**Review by fjlanasa — APPROVED** [review]
LGTM assuming you've tested out on Semaphore

**Comment by lemald** [comment]
Yup, this branch builds just fine with the configuration I put into Semaphore, and the caching of the Dialyzer stuff seems to work (or at least it doesn't take forever to run like it did the first time).

**Pull request merged**